http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/RuntimeV2ServiceInterpSpec.scala

```diff
@@ -1505,285 +1582,450 @@ class RuntimeV2ServiceInterpSpec extends AnyFlatSpec with LeonardoTestSuite with
-                            WorkbenchUserId("userId"),
-                            WorkbenchEmail("user1@example.com"),
-                            0
-    ) // this email is allowlisted
+    val userInfo = UserInfo(OAuth2BearerToken(""), WorkbenchUserId("userId"), WorkbenchEmail("user1@example.com"), 0)
+
+    val runtimeId1 = UUID.randomUUID.toString
+    val runtimeId2 = UUID.randomUUID.toString
+    val projectIdGcp = cloudContextGcp.asString
+    val workspaceIdAzure = UUID.randomUUID.toString
+
+    val mockAuthProvider = mockAuthorize(
+      userInfo,
+      List(RuntimeSamResourceId(runtimeId1), RuntimeSamResourceId(runtimeId2)),
+      List(WorkspaceResourceSamResourceId(WorkspaceId(UUID.fromString(workspaceIdAzure)))),
+      List(ProjectSamResourceId(GoogleProject(projectIdGcp)))
+    )
+
+    val testService = makeInterp(authProvider = mockAuthProvider)
 
     val res = for {
-      samResource1 <- IO(RuntimeSamResourceId(UUID.randomUUID.toString))
-      samResource2 <- IO(RuntimeSamResourceId(UUID.randomUUID.toString))
-      _ <- IO(makeCluster(1).copy(samResource = samResource1).save())
-      _ <- IO(
+      samResource1 <- IO(RuntimeSamResourceId(runtimeId1))
+      samResource2 <- IO(RuntimeSamResourceId(runtimeId2))
+      // GCP runtime
+      runtime1 <- IO(makeCluster(1).copy(samResource = samResource1, workspaceId = workspaceIdOpt).save())
+      // Azure runtime
+      runtime2 <- IO(
         makeCluster(2)
-          .copy(samResource = samResource2, cloudContext = CloudContext.Azure(CommonTestData.azureCloudContext))
+          .copy(
+            samResource = samResource2,
+            cloudContext = CloudContext.Azure(CommonTestData.azureCloudContext),
+            workspaceId = Some(WorkspaceId(UUID.fromString(workspaceIdAzure)))
+          )
           .save()
       )
-      listResponse <- runtimeV2Service.listRuntimes(userInfo, None, None, Map.empty)
-    } yield listResponse.map(_.samResource).toSet shouldBe Set(samResource1, samResource2)
+      listResponse <- testService.listRuntimes(userInfo, None, None, Map.empty)
+    } yield {
+      listResponse.map(_.samResource).toSet shouldBe Set(samResource1, samResource2)
+      listResponse.head shouldBe ListRuntimeResponse2(
+        id = runtime1.id,
+        workspaceId = workspaceIdOpt,
+        samResource = runtime1.samResource,
+        clusterName = runtime1.runtimeName,
+        cloudContext = runtime1.cloudContext,
+        auditInfo = runtime1.auditInfo,
+        runtimeConfig = defaultDataprocRuntimeConfig,
+        proxyUrl =
+          Runtime.getProxyUrl(proxyUrlBase, cloudContextGcp, runtime1.runtimeName, Set(jupyterImage), None, Map.empty),
+        runtime1.status,
+        runtime1.labels,
+        runtime1.patchInProgress
+      )
+    }
 
     res.unsafeRunSync()(cats.effect.unsafe.IORuntime.global)
   }
 
-  it should "fail to list runtimes if user loses access to workspace" taggedAs SlickPlainQueryTest in isolatedDbTest {
```

Comment:
is this case covered by the tests you added? (user is a creator of runtimes, but is then removed from a workspace) I've looked through, but maybe I missed something in the diff

Reply:
That should be covered by this case in the table:
```
  it should "list runtimes given different user permissions" taggedAs SlickPlainQueryTest in isolatedDbTest {
    forAll(
      Table(
        ("context", "runtimeAccess", "contextAccess", "isListed"),
        // Any runtime access; no access to wrapper context => hidden
		...
        (TestContext.GoogleProject, TestRuntimeAccess.Reader, TestContextAccess.Nothing, false),
        (TestContext.GoogleWorkspace, TestRuntimeAccess.Reader, TestContextAccess.Nothing, false),
        (TestContext.AzureWorkspace, TestRuntimeAccess.Reader, TestContextAccess.Nothing, false),
```